src/components/UserList.jsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState, useContext } from 'react';
+import BeatLoader from 'react-spinners/BeatLoader';
+import { UserContext } from '../contexts/UserContext';
+
+export default function UserList({ userList }) {
+  const [users, setUsers] = useState([]);
+  const [isLoading, setIsLoading] = useState();
+  const { setUser } = useContext(UserContext);
+  const [selectedUser, setSelectedUser] = useState();
+
+  useEffect(() => {
+    setIsLoading(true);
+    setUsers(userList.users);
```

Comment:
Can we put the set loading state to false inside a then block to ensure our asynchronous code runs at the correct.